src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -233,8 +234,7 @@ private static unsafe object[] InitializeStatics(IntPtr gcStaticRegionStart, int
                     Unsafe.Add(ref rawSpineData, currentBase) = obj;
 
                     // Update the base pointer to point to the pinned object
-                    *pBlock = *(IntPtr*)&obj;
-#pragma warning restore CS8500
+                    *pBlock = Unsafe.GetPinnedObjectPointer(obj);
```

Comment:
I do not think that this is going to build.

This is very special low-level code. I think what you have here earlier was better.